src/main/java/org/jabref/logic/citationstyle/CSLAdapter.java

```diff
@@ -82,15 +83,26 @@ private void initialize(String newStyle, CitationStyleOutputFormat newFormat) th
 
         /**
          * Converts the {@link BibEntry} into {@link CSLItemData}.
+         *
+         * Clone bibEntry to make a save changes,
+         * Change month field from JabRefFormat <code>#mon#</> to ShortName <code>mon</code>
+         * because CSL does not support JabRefFormat.
          */
         private static CSLItemData bibEntryToCSLItemData(BibEntry bibEntry) {
+
+            bibEntry = (BibEntry) bibEntry.clone();
+
             String citeKey = bibEntry.getCiteKeyOptional().orElse("");
             BibTeXEntry bibTeXEntry = new BibTeXEntry(new Key(bibEntry.getType()), new Key(citeKey));
 
             // Not every field is already generated into latex free fields
             HTMLChars latexToHtmlConverter = new HTMLChars();
             RemoveNewlinesFormatter removeNewlinesFormatter = new RemoveNewlinesFormatter();
             for (String key : bibEntry.getFieldMap().keySet()) {
+                if (FieldName.MONTH.equals(key)) {
+                    bibEntry.getFieldMap().put(FieldName.MONTH, bibEntry.getMonth().get().getShortName());
+                }
+
                 bibEntry.getField(key)
                         .map(removeNewlinesFormatter::format)
                         .map(latexToHtmlConverter::format)
```

Comment:
The code changes above work but are more complicated than they need to. I think an easier solution is to extract the lambda expression in this line (`ifPresent(value -> ...)`) to a new method `setFieldValue(bibTeXEntry, key, value)`. In this new method, you can then simply check `if (FieldName.MONTH.equals(key)) {` and use `Month.parse` to get the correct month value.

Reply:
I did not notice that bibEntry and BibTeXEntry are different object. I think we do not need to clone bibEntry anymore. Instead of create a new method I thought of making test direct on lamda expression. What do you think?

I tested this code and it work.

```
private static CSLItemData bibEntryToCSLItemData(BibEntry bibEntry) {
    String citeKey = bibEntry.getCiteKeyOptional().orElse("");
    BibTeXEntry bibTeXEntry = new BibTeXEntry(new Key(bibEntry.getType()), new Key(citeKey));

    // Not every field is already generated into latex free fields
    HTMLChars latexToHtmlConverter = new HTMLChars();
    RemoveNewlinesFormatter removeNewlinesFormatter = new RemoveNewlinesFormatter();
    for (String key : bibEntry.getFieldMap().keySet()) {
        bibEntry.getField(key)
                .map(removeNewlinesFormatter::format)
                .map(latexToHtmlConverter::format)
                .ifPresent(value -> {
                    if((FieldName.MONTH.equals(key)) && (value.matches("#[a-z]{3}#"))){
                        value = value.substring(1,4);
                    }
                    bibTeXEntry.addField(new Key(key), new DigitStringValue(value));
                });
    }
    return BIBTEX_CONVERTER.toItemData(bibTeXEntry);
}
```

About https://github.com/JabRef/jabref/blob/master/src/test/java/org/jabref/logic/citationstyle/CitationStyleTest.java I have no idea what kind of test I will make because I never make tests for my programs.

Reply:
That is also an idea, but I would not make the regex. If you have the field month, I would really get the shortname from the BibEntry then, so value = key.getMonth().getShortName

Reply:
```for (String key : bibEntry.getFieldMap().keySet()) {
    bibEntry.getField(key)
            .map(removeNewlinesFormatter::format)
            .map(latexToHtmlConverter::format)
            .ifPresent(value -> {
                if((FieldName.MONTH.equals(key))){
                    value = bibEntry.getMonth().get().getShortName();
                }
                bibTeXEntry.addField(new Key(key), new DigitStringValue(value));
            });
}
```

Reply:
```
for (String key : bibEntry.getFieldMap().keySet()) {
    bibEntry.getField(key)
            .map(removeNewlinesFormatter::format)
            .map(latexToHtmlConverter::format)
            .ifPresent(value -> {
                if(FieldName.MONTH.equals(key)){
                    value = bibEntry.getMonth().get().getShortName();
                }
                bibTeXEntry.addField(new Key(key), new DigitStringValue(value));
            });
}
```

Reply:
I just post my last request